### ISSUE TITLE
font validation failed, installing Meslo font

### ISSUE THREAD

**Issue by zno5** [ryanoasis/nerd-fonts#413]
#### 🎯 font validation failed

Font validation has failed when I install Meslo bold italic font.
Other fonts(bold, italic, regular) are fine.

The problem is name table usability has caution.

#### 🔧 Your Setup

- Meslo LG S Bold Italic Nerd Font Complete Mono.ttf
- macOS 10.14.6

#### ★ Optional

<img width="734" alt="Screen Shot 2019-12-28 at 10 37 20 AM" src="https://user-images.githubusercontent.com/2455180/71537345-0a053700-295e-11ea-9209-3b93b3ef38af.png">

**Comment by ryanoasis** [comment]
@zno5 Thanks for the report, low priority issue but issue non-the-less. Is it the same warning with regular Meslo font (non Nerd Font patched) ?

**Comment by yaneura-no-gomi** [comment]
I have the same proble with Meslo LG M Bold Nerd Font Complete Mono.ttf

**Comment by yaneura-no-gomi** [comment]
I can get Bold font from https://www.nerdfonts.com/font-downloads
thank you !

**Comment by PatTheMav** [comment]
@ryanoasis Fwiw I had the same issue when patching SF Mono recently (which felt like a regression because I didn't use to have that issue). What "fixed" it for me was to add this little stinker before the `subFamily` is added to `fontname`:

```
        subFamily = subFamily.replace(' ', '')
```

Whereas the `fontname` was `SFMonoNerdFont-Regular Italic`, it now became `SFMonoNerdFont-RegularItalic` - without any spaces in there (they are removed from the `fontname` as well) macOS seems to like the name table again.

To summarise:

*Results in bad name table*:

```
Family Name:    SFMono Nerd Font
Full Name:      SFMono Regular Italic Nerd Font
Font Name:      SFMonoNerdFont-Regular Italic
Sub-Family:     Regular Italic
```

*Results in good name table*:

```
Family Name:    SFMono Nerd Font
Full Name:      SFMono Regular Italic Nerd Font
Font Name:      SFMonoNerdFont-RegularItalic
Sub-Family:     RegularItalic
```

**Comment by torarnv** [comment]
Had this after patching Menlo:

<img width="752" alt="image" src="https://user-images.githubusercontent.com/7536/162619830-898bfb84-8d0f-4428-9ecc-99d71263fbe8.png">

**Comment by torarnv** [comment]
The ` subFamily = subFamily.replace(' ', '')` trick worked.

**Comment by torarnv** [comment]
Fixed in #820. Only the font name (PS name) needs sanitation.
